### PR TITLE
Add support for passing in paths as an array which aren't split on

### PR DESCRIPTION
**Pull request by josephschmitt**
This allows using keys that have periods in them.

### CODE REVIEW

**Comment by acstll** [comment]
Thank you @josephschmitt for this PR (specially for the README and adding tests!).

I really like this array syntax, plus it's a non-breaking change.

I found a couple of errors (typos actually) on the tests that prevent them from passing. I will add comments here so you can fix them. I can then review the whole thing before merging. (Please let me know if you prefer that I make these changes.)

Thanks again.

File: test.js [hunk 1]
```diff
@@ -17,6 +17,23 @@ test('deep gets', function (t) {
...
+      baz: {
+        beep: 'boop'
+      },
+      'baz.beep': 'blop'
+    }
+  };
+  
+  t.equal(deep(obj, ['bar', 'baz', 'beep']), 'boop');
+  t.equal(deep(obj, ['bar', 'baz', 'beep', 'yep', 'nope']), undefined);
+  t.equal(deep(obj, ['bar', 'bar.beep']), 'blop');
```
Comment (acstll) [inline test.js]: On line 33

```diff
- t.equal(deep(obj, ['bar', 'bar.beep']), 'blop');
+ t.equal(deep(obj, ['bar', 'baz.beep']), 'blop');
```

File: test.js [hunk 2]
```diff
@@ -45,6 +62,25 @@ test('deep sets', function (t) {
   t.end();
 });
 
+test('deep sets with array of paths', function (r) {
```
Comment (acstll) [inline test.js]: On line 65

```diff
- test('deep sets with array of paths', function (r) {
+ test('deep sets with array of paths', function (t) {
```

File: test.js [hunk 3]
```diff
@@ -45,6 +62,25 @@ test('deep sets', function (t) {
...
+      }
+    }
+  };
+
+  t.equal(deep(obj, 'foo', 'yep'), 'yep');
+  t.equal(obj.foo, 'yep');
+  t.equal(deep(obj, ['bar', 'baz', 'beep'], 'nope'), 'nope');
+  t.equal(obj.bar.baz.beep, 'nope');
+  t.equal(deep(obj, ['bar', 'baz.beep'], 'nooope'), 'nooope');
+  t.equal(obj.bar.['baz.beep'], 'nooope');
```
Comment (acstll) [inline test.js]: On line 80

```diff
- t.equal(obj.bar.['baz.beep'], 'nooope');
+ t.equal(obj.bar['baz.beep'], 'nooope');
```

**Comment by acstll** [comment]
After thinking about it for a bit, adding `Array.isArray` could be a breaking change, so I feel better bumping the major number. Let me know what you think…

**Comment by josephschmitt** [comment]
Whoops! Thanks for catching those. New commit is up that fixes them. I think a 1.1.0 makes sense here, especially since it's an API change (though mostly an invisible, backwards-compatible one). Want me to bump it?

**Comment by acstll** [comment]
Thanks!

Let's make it 1.1.0 then (I just checked `Array.isArray` is widely available …I remember it being a problem). Yes, please bump it if you're so kind.

**Comment by josephschmitt** [comment]
@acstll all set

**Comment by acstll** [comment]
@josephschmitt published. Thanks a lot!

**Comment by josephschmitt** [comment]
Thanks for being so quick to accept a PR!

**Pull request merged**